src/shared/ui/inputPrompter.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { applyPrimitives } from '../utilities/collectionUtils'
+import { WIZARD_BACK } from '../wizards/wizard'
+import { QuickInputButton, PrompterButtons } from './buttons'
+import { Prompter, PromptResult } from './prompter'
+
+/** Additional options to configure the `InputBox` beyond the standard API */
```

Comment:
It's to get the types to work how they need to work. But since this causes indirection when trying to view the interface I'm just going to define the interface directly.